**Crash-Reporting Drivers Data Analysis** 

This dataset provides information on motor vehicle operators (drivers) involved in traffic collisions occurring on county and local roadways. The dataset reports details of all traffic collisions occurring on county and local roadways within Montgomery County, as collected via the Automated Crash Reporting System (ACRS) of the Maryland State Police, and reported by the Montgomery County Police, Gaithersburg Police, Rockville Police, or the Maryland-National Capital Park Police. This dataset shows each collision data recorded and the drivers involved.  

<span style="color: rgb(27, 27, 27); font-family: Lato, Arial, sans-serif;"><br></span>

"I have downloaded this data from **data.gov**, and it pertains to the **federal government**, specifically the **Montgomery County of Maryland.** I intend to use **SQL** to derive insights from this dataset."

**Content**

**This dataset contains 171420 rows and 40 columns**

**Below are the variable descriptions for each of the columns:**

**Local Case Number:** A unique identifier for a specific case related to a crash or incident.

**Agency Name:** The name or identifier of the agency involved or responsible for handling the case.

**ACRS Report Type:** The categorization or type of the Accident Crash Reporting System (ACRS) report.

**Crash Date/Time:** The date and time when the crash or incident occurred.

**Route Type:** The type or classification of the route where the incident occurred.

**Road Name:** The name of the road where the incident occurred.

**Cross-Street Type:** The type or classification of the intersecting street.

**Cross-Street Name:** The name of the intersecting street.

**Off-Road Description:** Description of the location or context, particularly when off the main road.

**Municipality:** The local government area or jurisdiction where the incident took place.

**Related Non-Motorist:** Information regarding any non-motorists involved in the incident.

**Collision Type:** The type or nature of the collision that occurred.

**Weather:** Weather conditions prevailing at the time of the incident.

**Surface Condition:** Description of the surface condition at the location of the incident.

**Light:** Information about lighting conditions during the incident.

**Traffic Control:** Details regarding traffic control measures at the location.

**Driver Substance Abuse:** Any indication or information about substance abuse by the driver.

**Non-Motorist Substance Abuse**: Information about substance abuse by non-motorists involved.

**Person ID:** Unique identifier for individuals involved in the incident.

**Driver At Fault:** Indicates whether the driver was at fault.

**Injury Severity:** Severity of injuries sustained as a result of the incident.

**Circumstance:** Circumstances or contextual information related to the incident.

**Driver Distracted By:** Details about what the driver was distracted by, if applicable.

**Drivers License State:** State of the driver's license.

**Vehicle ID:** Unique identifier for the vehicle involved in the incident.

**Vehicle Damage Extent:** Extent of damage sustained by the vehicle.

**Vehicle First Impact Location:** Location on the vehicle where the first impact occurred.

**Vehicle Second Impact Location:** Location on the vehicle where the second impact occurred.

**Vehicle Body Type:** Type or classification of the vehicle's body.

**Vehicle Movement:** Description of the vehicle's movement at the time of the incident.

**Vehicle Continuing Dir:** Direction of the vehicle's continued movement post-incident.

**Vehicle Going Dir:** Direction the vehicle was heading at the time of the incident.

**Speed Limit:** Speed limit applicable at the incident location.

**Driverless Vehicle:** Indicates whether the vehicle was driverless.

**Parked Vehicle:** Indicates whether the vehicle was parked.

**Vehicle Year:** Year of manufacture of the vehicle.

**Vehicle Make:** Manufacturer of the vehicle.

**Vehicle Model:** Model of the vehicle.

**Equipment Problems:** Information about any equipment-related issues observed.

### **Importing the Data into PostgreSql**

In [ ]:
DROP TABLE IF EXISTS Crash_Reporting;

CREATE TABLE Crash_Reporting(
	"Report Number" VARCHAR,
	"Local Case Number" VARCHAR,
	"Agency Name" VARCHAR,
	"ACRS Report Type" VARCHAR,
	"Crash Date/Time" VARCHAR,
	"Route Type" VARCHAR,
	"Road Name" VARCHAR,
	"Cross-Street Type" VARCHAR,
	"Cross-Street Name" VARCHAR,
	"Off-Road Description" VARCHAR,
	Municipality VARCHAR,
	"Related Non-Motorist" VARCHAR,
	"Collision Type" VARCHAR,
	Weather VARCHAR,
	"Surface Condition" VARCHAR,
	Light VARCHAR,
	"Traffic Control" VARCHAR,
	"Driver Substance Abuse" VARCHAR,
	"Non-Motorist Substance Abuse" VARCHAR,
	"Person ID" VARCHAR,
	"Driver At Fault" VARCHAR,
	"Injury Severity" VARCHAR,
	Circumstance VARCHAR,
	"Driver Distracted By" VARCHAR,
	"Drivers License State" VARCHAR,
	"Vehicle ID" VARCHAR,
	"Vehicle Damage Extent" VARCHAR,
	"Vehicle First Impact Location" VARCHAR,
	"Vehicle Second Impact Location" VARCHAR,
	"Vehicle Body Type" VARCHAR,
	"Vehicle Movement" VARCHAR,
	"Vehicle Continuing Dir" VARCHAR,
	"Vehicle Going Dir" VARCHAR,
	"Speed Limit" INT,
	"Driverless Vehicle" VARCHAR,
	"Parked Vehicle" VARCHAR,
	"Vehicle Year" INT,
	"Vehicle Make" VARCHAR,
	"Vehicle Model" VARCHAR,
	"Equipment Problems" VARCHAR
	
	);
	
	--Although I have created table as Crash_Reporting ,so further I am using this only

In [24]:
--Look at first few records only 

Select *
from Crash_Reporting 
Limit 10;

(10 row(s) affected)

Total execution time: 00:00:00.010

Report Number Local Case Number Agency Name ACRS Report Type Crash Date/Time Route Type Road Name Cross-Street Type Cross-Street Name Off-Road Description municipality Related Non-Motorist Collision Type weather Surface Condition light Traffic Control Driver Substance Abuse Non-Motorist Substance Abuse Person ID Driver At Fault Injury Severity circumstance Driver Distracted By Drivers License State Vehicle ID Vehicle Damage Extent Vehicle First Impact Location Vehicle Second Impact Location Vehicle Body Type Vehicle Movement Vehicle Continuing Dir Vehicle Going Dir Speed Limit Driverless Vehicle Parked Vehicle Vehicle Year Vehicle Make Vehicle Model Equipment Problems MCP3040003N 190026050 Montgomery County Police Property Damage Crash 05/31/2019 03:00:00 PM NULL NULL NULL NULL PARKING LOT OF 3215 SPARTAN RD NULL NULL OTHER CLEAR NULL DAYLIGHT N/A UNKNOWN NULL DE2A24CD-7919-4F8D-BABF-5B75CE12D21E Yes NO APPARENT INJURY N/A UNKNOWN NULL 165AD539-A8C8-4004-AF73-B7DCAAA8B3CC SUPERFICIAL ONE OCLOCK ONE OCLOCK PASSENGER CAR PARKING North North 15 No No 2004 HONDA TK UNKNOWN EJ78850038 230034791 Gaithersburg Police Depar Property Damage Crash 07/21/2023 05:59:00 PM Maryland (State) FREDERICK RD Unknown WATKINS MILL RD NULL N/A NULL STRAIGHT MOVEMENT ANGLE CLEAR DRY DAYLIGHT TRAFFIC SIGNAL NONE DETECTED NULL E7058A8E-4F18-4D2A-954E-04A099CFED12 No NO APPARENT INJURY N/A NOT DISTRACTED MD 1C3C3E2F-9A23-4ED0-9BB3-B6C370D99C37 DISABLING THREE OCLOCK TWO OCLOCK PASSENGER CAR MAKING LEFT TURN East South 40 No No 2011 GMC TK NO MISUSE MCP2009002G 230034583 Montgomery County Police Property Damage Crash 07/20/2023 03:10:00 PM Maryland (State) GEORGIA AVE Maryland (State) NORBECK RD NULL N/A NULL STRAIGHT MOVEMENT ANGLE CLEAR DRY DAYLIGHT TRAFFIC SIGNAL NONE DETECTED NULL 2B404D6D-8DB5-4CB6-9E71-9F1B8D0A8925 Yes NO APPARENT INJURY N/A NOT DISTRACTED MD 0483CE47-E0FC-4BCA-BAB0-B7541820FEE6 FUNCTIONAL TWELVE OCLOCK TWELVE OCLOCK PICKUP TRUCK ACCELERATING North North 35 No No 2019 FORD F150 NO MISUSE MCP3201004C 230035036 Montgomery County Police Property Damage Crash 07/23/2023 12:10:00 PM County CRYSTAL ROCK DR County WATERS LANDING DR NULL N/A NULL STRAIGHT MOVEMENT ANGLE CLEAR DRY DAYLIGHT NO CONTROLS NONE DETECTED NULL 637D8107-0381-4B8D-848A-B4A93B4D53CE Yes NO APPARENT INJURY N/A LOOKED BUT DID NOT SEE MD 4406AA84-07F8-45F4-88A2-09AD89AC9AAF FUNCTIONAL TWELVE OCLOCK TWELVE OCLOCK PASSENGER CAR STARTING FROM LANE East East 40 No No 2016 KIA SW NO MISUSE MCP23290028 230035152 Montgomery County Police Property Damage Crash 07/24/2023 06:10:00 AM County MONTGOMERY VILLAGE AVE County CENTERWAY RD NULL N/A NULL HEAD ON LEFT TURN CLOUDY DRY DAYLIGHT TRAFFIC SIGNAL NONE DETECTED NULL 52F904CB-822C-4D8D-9457-0D66B99AD407 Yes NO APPARENT INJURY N/A NOT DISTRACTED CA 148FAA0F-4EBB-43E8-9339-BBED3F245751 FUNCTIONAL TWELVE OCLOCK TWELVE OCLOCK (SPORT) UTILITY VEHICLE MAKING LEFT TURN West North 35 No No 2016 TOYT TK NO MISUSE MCP295200DV 230032956 Montgomery County Police Property Damage Crash 07-11-2023 07:40:00 AM County WAYNE AVE County ALLEY NULL N/A NULL SAME DIR REAR END CLEAR DRY DAYLIGHT NO CONTROLS NONE DETECTED NULL 800B57CA-3277-4DC5-8C8D-457AF042F683 No NO APPARENT INJURY N/A NOT DISTRACTED MD E9504EBF-592D-4E6D-B28E-07B8773E2530 FUNCTIONAL SIX OCLOCK SIX OCLOCK TRANSIT BUS STOPPED IN TRAFFIC LANE East East 30 No No 2016 GILL BUS NO MISUSE MCP33510013 230033282 Montgomery County Police Property Damage Crash 07-12-2023 08:28:00 PM Maryland (State) COLESVILLE RD US (State) COLESVILLE RD NULL N/A NULL SAME DIR REAR END CLEAR DRY DUSK NO CONTROLS NONE DETECTED NULL 3F9D8C7E-6F06-45DC-8250-E30A8ACC5F2A Yes NO APPARENT INJURY ANIMAL, N/A NOT DISTRACTED MD 870FF895-E20E-41E0-BE96-CC7D86335B37 DISABLING TWELVE OCLOCK TWELVE OCLOCK (SPORT) UTILITY VEHICLE SLOWING OR STOPPING South South 25 No No 2014 SUBA SW NO MISUSE EJ7869003F 230032124 Gaithersburg Police Depar Injury Crash 07-05-2023 11:25:00 PM Maryland (State) CLOPPER RD Unknown STRUC #15093

**What is the total count of recorded crashes in the complete dataset ?**

In [25]:
Select COUNT(*)
from Crash_Reporting;

(1 row(s) affected)

Total execution time: 00:00:00.134

count 171419

**What is the earliest and latest date of recorded crashes ?**

In [18]:
SELECT 
    MIN(TO_TIMESTAMP("Crash Date/Time", 'MM-DD-YYYY HH24:MI:SS')) AS Earliest_Date,
    MAX(TO_TIMESTAMP("Crash Date/Time", 'MM-DD-YYYY HH24:MI:SS')) AS Latest_Date
FROM 
    Crash_Reporting;


(1 row(s) affected)

Total execution time: 00:00:00.615

earliest_date latest_date 2015-01-01 01:10:00+05:30 2023-12-20 12:21:00+05:30

**What is the number of reported crashes Per Year ?**

In [38]:
SELECT EXTRACT(YEAR FROM TO_TIMESTAMP("Crash Date/Time", 'MM-DD-YYYY HH24:MI:SS')) AS Crash_Year,
Count(*) as Reported_Crashes
FROM Crash_Reporting
GROUP BY EXTRACT(YEAR FROM TO_TIMESTAMP("Crash Date/Time", 'MM-DD-YYYY HH24:MI:SS'))
ORDER BY Crash_Year;

(9 row(s) affected)

Total execution time: 00:00:00.899

crash_year reported_crashes 2015 20286 2016 21778 2017 21541 2018 21042 2019 20941 2020 13809 2021 16206 2022 17578 2023 18238

**The table indicates that there were fewer crash reports in the year 2023, while the year 2016 witnessed a higher number of crash reports.**

**Find the counts of Accidents in each Municipality(**<span style="color: var(--vscode-foreground);">The local government area or jurisdiction where the incident took place)</span> **?**

In [39]:
Select Municipality,COUNT(*) as NumAccidents
from Crash_Reporting
GROUP by Municipality;

(22 row(s) affected)

Total execution time: 00:00:00.407

municipality numaccidents BROOKEVILLE 16 CHEVY CHASE #3 156 CHEVY CHASE #4 325 CHEVY CHASE #5 74 CHEVY CHASE VIEW 93 CHEVY CHASE VILLAGE 87 DRUMMOND 12 FRIENDSHIP HEIGHTS 133 GAITHERSBURG 6402 GARRETT PARK 51 GLEN ECHO 33 KENSINGTON 411 LAYTONSVILLE 25 MATINS ADDITION 28 N/A 136501 NORTH CHEVY CHASE 34 POOLESVILLE 107 ROCKVILLE 9294 SOMERSET 53 TAKOMA PARK 1676 WASHINGTON GROVE 41 NULL 15867

**"Rockville, with 9294 accidents recorded, has the highest number of reported incidents, including cases with locations listed as null, which are not properly located."**

**Identify the most frequent collision Types in Accidents ?**

In [40]:
Select "Collision Type" ,COUNT(*) as CollisionCount
From Crash_Reporting
GROUP by "Collision Type"
Order by CollisionCount;

(19 row(s) affected)

Total execution time: 00:00:00.511

Collision Type collisioncount OPPOSITE DIR BOTH LEFT TURN 320 N/A 580 ANGLE MEETS LEFT HEAD ON 700 UNKNOWN 714 SAME DIR BOTH LEFT TURN 717 SAME DIR REND RIGHT TURN 720 SAME DIR REND LEFT TURN 739 ANGLE MEETS RIGHT TURN 1204 ANGLE MEETS LEFT TURN 2018 OPPOSITE DIRECTION SIDESWIPE 2867 SAME DIRECTION LEFT TURN 3701 HEAD ON 3766 SAME DIRECTION RIGHT TURN 3817 HEAD ON LEFT TURN 12883 SINGLE VEHICLE 15796 SAME DIRECTION SIDESWIPE 16161 OTHER 18947 STRAIGHT MOVEMENT ANGLE 30212 SAME DIR REAR END 55557

**List the total amount of Fatal Crash, Injury Crash and Property Damage Crash reported between 2015 and 2023.**

In [46]:
SELECT
    "ACRS Report Type" as Crime_Type,
    COUNT(*) as Number_crimes
FROM 
    Crash_Reporting
WHERE 
    "ACRS Report Type" IN('Fatal Crash','Injury Crash','Property Damage Crash')
GROUP BY Crime_Type
ORDER by Number_crimes;


(3 row(s) affected)

Total execution time: 00:00:00.406

crime_type number_crimes Fatal Crash 423 Injury Crash 62013 Property Damage Crash 108983

 **Find incidents involving driver distraction and then analyze the distribution of distraction reasons.**

In [36]:
WITH DistractedIncidents AS(
    SELECT *
    FROM Crash_Reporting
    WHERE "Driver Distracted By" is NOT NULL
)
SELECT "Driver Distracted By",Count(*) as Distraction_Count
From DistractedIncidents
GROUP BY "Driver Distracted By";

(19 row(s) affected)

Total execution time: 00:00:00.162

Driver Distracted By distraction_count ADJUSTING AUDIO AND OR CLIMATE CONTROLS 133 BY MOVING OBJECT IN VEHICLE 212 BY OTHER OCCUPANTS 418 DIALING CELLULAR PHONE 48 DISTRACTED BY OUTSIDE PERSON OBJECT OR EVENT 969 EATING OR DRINKING 195 INATTENTIVE OR LOST IN THOUGHT 4301 LOOKED BUT DID NOT SEE 21438 NO DRIVER PRESENT 280 NOT DISTRACTED 106317 OTHER CELLULAR PHONE RELATED 371 OTHER DISTRACTION 3236 OTHER ELECTRONIC DEVICE (NAVIGATIONAL PALM PILOT) 323 SMOKING RELATED 26 TALKING OR LISTENING TO CELLULAR PHONE 281 TEXTING FROM A CELLULAR PHONE 66 UNKNOWN 32650 USING DEVICE OBJECT BROUGHT INTO VEHICLE 63 USING OTHER DEVICE CONTROLS INTEGRAL TO VEHICLE 92

**Calculate the average speed limit of incidents involving different collision types.**

In [46]:
SELECT "Collision Type", 
       Round(AVG("Speed Limit"),2) as Average_SpeedLimit
FROM Crash_Reporting
GROUP BY "Collision Type";


(19 row(s) affected)

Total execution time: 00:00:00.166

Collision Type average_speedlimit ANGLE MEETS LEFT HEAD ON 31.01 ANGLE MEETS LEFT TURN 34.11 ANGLE MEETS RIGHT TURN 33.35 HEAD ON 29.99 HEAD ON LEFT TURN 35.09 N/A 23.40 OPPOSITE DIR BOTH LEFT TURN 29.97 OPPOSITE DIRECTION SIDESWIPE 25.90 OTHER 23.30 SAME DIR BOTH LEFT TURN 33.09 SAME DIR REAR END 36.26 SAME DIR REND LEFT TURN 35.12 SAME DIR REND RIGHT TURN 33.93 SAME DIRECTION LEFT TURN 33.91 SAME DIRECTION RIGHT TURN 33.50 SAME DIRECTION SIDESWIPE 32.68 SINGLE VEHICLE 30.58 STRAIGHT MOVEMENT ANGLE 32.46 UNKNOWN 16.16

**Count accidents associated with different weather conditions.**

In [47]:
Select 
    Weather,
    COUNT(*) as Accident_Count
FROM 
    Crash_Reporting
WHERE
    Weather is NOT NULL
GROUP by Weather
ORDER by Accident_Count;

(13 row(s) affected)

Total execution time: 00:00:00.239

weather accident_count BLOWING SAND, SOIL, DIRT 15 BLOWING SNOW 115 SEVERE WINDS 154 SLEET 218 OTHER 370 WINTRY MIX 389 FOGGY 668 UNKNOWN 684 SNOW 1443 N/A 13311 CLOUDY 17209 RAINING 20478 CLEAR 116365

**Determine the top vehicle makes and models involved in accidents.**

In [22]:
Select 
    "Vehicle Make",
    "Vehicle Model",
    COUNT(*) AS Vehicle_Count 
FROM 
    Crash_Reporting
Where 
    "Vehicle Make" is NOT NULL
    AND "Vehicle Model" is NOT NULL
GROUP BY 
    "Vehicle Make",
    "Vehicle Model"
ORDER BY 
    Vehicle_Count DESC
Limit 20;




    

(20 row(s) affected)

Total execution time: 00:00:00.243

Vehicle Make Vehicle Model vehicle_count TOYOTA CAMRY 5123 HONDA CIVIC 4979 HONDA ACCORD 4895 TOYOTA COROLLA 4656 UNKNOWN UNKNOWN 3646 TOYT 4S 2466 HONDA CRV 2379 NISSAN ALTIMA 2175 TOYOTA RAV4 2109 FORD TK 1912 HOND 4S 1608 FORD EXPLORER 1427 DODGE CHARGER 1419 TOYOTA PRIUS 1314 TOYOTA HIGHLANDER 1260 HONDA PILOT 1240 FORD F150 1228 TOYOTA SIENNA 1203 NISSAN SENTRA 1147 HONDA ODYSSEY 1122

**Analyze patterns in accidents based on crash date/time, e.g., high accident months or days.**

In [58]:
SELECT EXTRACT(MONTH FROM TO_TIMESTAMP("Crash Date/Time", 'MM-DD-YYYY HH24:MI:SS')) AS Accident_Month,
       COUNT(*) AS NumAccidents
FROM Crash_Reporting
GROUP BY Accident_Month
ORDER BY NumAccidents DESC;


(12 row(s) affected)

Total execution time: 00:00:01.167

accident_month numaccidents 10 16093 9 15135 11 14740 12 14521 5 14414 7 14389 8 14376 1 14140 6 13763 3 13488 4 13363 2 12997

**Investigate relationships between injury severity and Driver Distracted by**

In [1]:
Select "Injury Severity",
    COUNT(*) as NumAccidents,
    "Driver Distracted By" as Distraction_Type
From 
    Crash_Reporting
Where
    "Driver Distracted By" is NOT NULL
GROUP BY "Injury Severity",Distraction_Type
ORDER by NumAccidents DESC
Limit 15;

(15 row(s) affected)

Total execution time: 00:00:00.438

Injury Severity numaccidents distraction_type NO APPARENT INJURY 84997 NOT DISTRACTED NO APPARENT INJURY 27601 UNKNOWN NO APPARENT INJURY 18764 LOOKED BUT DID NOT SEE POSSIBLE INJURY 12370 NOT DISTRACTED SUSPECTED MINOR INJURY 8116 NOT DISTRACTED NO APPARENT INJURY 3604 INATTENTIVE OR LOST IN THOUGHT NO APPARENT INJURY 2660 OTHER DISTRACTION POSSIBLE INJURY 2533 UNKNOWN SUSPECTED MINOR INJURY 1962 UNKNOWN POSSIBLE INJURY 1592 LOOKED BUT DID NOT SEE SUSPECTED MINOR INJURY 995 LOOKED BUT DID NOT SEE NO APPARENT INJURY 840 DISTRACTED BY OUTSIDE PERSON OBJECT OR EVENT SUSPECTED SERIOUS INJURY 820 NOT DISTRACTED SUSPECTED SERIOUS INJURY 417 UNKNOWN POSSIBLE INJURY 370 INATTENTIVE OR LOST IN THOUGHT

**Most Common Cross-Street Types in Accidents with Parked Vehicles**

In [12]:
SELECT "Cross-Street Type",
       COUNT(*) AS Num_Accidents
FROM Crash_Reporting
WHERE "Parked Vehicle" = 'Yes'
GROUP BY "Cross-Street Type"
ORDER BY Num_Accidents DESC;


(11 row(s) affected)

Total execution time: 00:00:00.261

Cross-Street Type num_accidents NULL 1482 County 706 Municipality 187 Maryland (State) 136 Unknown 89 US (State) 25 Other Public Roadway 22 Ramp 13 Government 8 Interstate (State) 3 Service Road 1

**What are the different types of lighting conditions and the number of crashes?**

In [5]:
Select 
    DISTINCT Light,
    COUNT(*) as Num_Accidents
FROM Crash_Reporting
GROUP by Light
ORDER by Num_Accidents;

(9 row(s) affected)

Total execution time: 00:00:00.326

light num_accidents OTHER 340 UNKNOWN 675 N/A 1441 DARK -- UNKNOWN LIGHTING 1565 DAWN 3439 DUSK 3916 DARK NO LIGHTS 4929 DARK LIGHTS ON 39287 DAYLIGHT 115827

**What are the top 10 deadliest streets?**

In [23]:
SELECT
    "Road Name",
    COUNT("Injury Severity") AS TotalFatalities
FROM
    Crash_Reporting
GROUP BY
    "Road Name"
ORDER BY
    TotalFatalities DESC
LIMIT 10;


(10 row(s) affected)

Total execution time: 00:00:00.159

Road Name totalfatalities NULL 15867 GEORGIA AVE 10825 NEW HAMPSHIRE AVE 6797 FREDERICK RD 5744 ROCKVILLE PIKE 4825 CONNECTICUT AVE 4083 VEIRS MILL RD 3967 COLUMBIA PIKE 3790 RANDOLPH RD 3438 COLESVILLE RD 3373

**Find  top 10 incidents where the weather was 'Rain' and the speed limit was greater than 40 mph.**

In [54]:
SELECT *
FROM Crash_Reporting
WHERE "weather" = 'RAINING'
  AND "Speed Limit" > 40
Limit 10;


(10 row(s) affected)

Total execution time: 00:00:00.005

Report Number Local Case Number Agency Name ACRS Report Type Crash Date/Time Route Type Road Name Cross-Street Type Cross-Street Name Off-Road Description municipality Related Non-Motorist Collision Type weather Surface Condition light Traffic Control Driver Substance Abuse Non-Motorist Substance Abuse Person ID Driver At Fault Injury Severity circumstance Driver Distracted By Drivers License State Vehicle ID Vehicle Damage Extent Vehicle First Impact Location Vehicle Second Impact Location Vehicle Body Type Vehicle Movement Vehicle Continuing Dir Vehicle Going Dir Speed Limit Driverless Vehicle Parked Vehicle Vehicle Year Vehicle Make Vehicle Model Equipment Problems MCP3280003P 230035153 Montgomery County Police Property Damage Crash 07/24/2023 05:50:00 AM US (State) COLUMBIA PIKE Unknown END BRIDGE NULL N/A NULL SAME DIR REAR END RAINING WET DARK NO LIGHTS NO CONTROLS UNKNOWN NULL 4D15D7FA-8D8D-4698-B68E-71B7AFBFA576 Yes NO APPARENT INJURY RAIN, SNOW, WET UNKNOWN NULL CCAC88E1-FC89-46E4-BF88-D0ABBC6BB592 UNKNOWN SIX OCLOCK SIX OCLOCK UNKNOWN CHANGING LANES North North 50 No No 0 UNKNOWN UNKNOWN UNKNOWN MCP3280003P 230035153 Montgomery County Police Property Damage Crash 07/24/2023 05:50:00 AM US (State) COLUMBIA PIKE Unknown END BRIDGE NULL N/A NULL SAME DIR REAR END RAINING WET DARK NO LIGHTS NO CONTROLS NONE DETECTED NULL AC6C890E-4FEE-4CCF-BBC5-6E9ED17C7FB5 No NO APPARENT INJURY RAIN, SNOW, WET NOT DISTRACTED MD 3E473103-06A4-4230-8EEB-19CC4AAC46E0 FUNCTIONAL TWELVE OCLOCK TWELVE OCLOCK TRANSIT BUS MOVING CONSTANT SPEED North North 50 No No 2010 NEW FLYER LOW FLOOR NO MISUSE MCP3129007H 230047738 Montgomery County Police Property Damage Crash 04/08/2023 06:24:00 AM Maryland (State) VEIRS MILL RD County ASPEN HILL RD NULL N/A NULL SINGLE VEHICLE RAINING WET DAWN NO CONTROLS NONE DETECTED NULL A894F676-BDBB-46B5-88D8-399169158F4E Unknown NO APPARENT INJURY RAIN, SNOW, WET NOT DISTRACTED MD ED83145D-B209-469D-AA1D-E83FFF7793EF FUNCTIONAL FOUR OCLOCK FOUR OCLOCK PASSENGER CAR MAKING RIGHT TURN West West 45 No No 2004 LINCOLN TOWN CAR NO MISUSE MCP2966006R 230046514 Montgomery County Police Property Damage Crash 07/27/2023 06:23:00 PM Interstate (State) EISENHOWER MEMORIAL HWY Unknown MILE MARKER 20.0 NULL N/A NULL SAME DIR REAR END RAINING N/A N/A NO CONTROLS NONE DETECTED NULL 3AE279B3-FEDB-4365-B5B3-B5780C2411E8 Yes NO APPARENT INJURY N/A, RAIN, SNOW NOT DISTRACTED VA A96F06FA-DB2E-410A-AEB8-72720906C787 FUNCTIONAL TWELVE OCLOCK TWELVE OCLOCK PASSENGER CAR MOVING CONSTANT SPEED North North 65 No No 2023 CHRYSLER PACIFICA N/A MCP2966006R 230046514 Montgomery County Police Property Damage Crash 07/27/2023 06:23:00 PM Interstate (State) EISENHOWER MEMORIAL HWY Unknown MILE MARKER 20.0 NULL N/A NULL SAME DIR REAR END RAINING N/A N/A NO CONTROLS NONE DETECTED NULL 4C5C5B00-4C95-4590-9588-7C2B8289EBB1 No NO APPARENT INJURY N/A, RAIN, SNOW NOT DISTRACTED MD 57192CD6-B841-45D6-8D5F-DB031BBFC6FE SUPERFICIAL SIX OCLOCK SIX OCLOCK POLICE VEHICLE/NON EMERGENCY SLOWING OR STOPPING North North 65 No No 2018 CHEVROLET SUBURBAN N/A MCP3301001S 230048145 Montgomery County Police Injury Crash 06/08/2023 10:21:00 PM Maryland (State) SANDY SPRING RD County BIRMINGHAM DR NULL N/A NULL ANGLE MEETS LEFT TURN RAINING WET DARK LIGHTS ON NO CONTROLS NONE DETECTED NULL E263A8B7-5614-4252-8EF0-F4F812D141FD Yes SUSPECTED MINOR INJURY RAIN, SNOW, WET UNKNOWN MD A5B22CDB-65F6-4D62-BA95-9D880E2BC219 DISABLING TWO OCLOCK TWO OCLOCK PASSENGER CAR MAKING U TURN West West 45 No No 2015 CHEV 4D NO MISUSE MCP3301001S 230048145 Montgomery County Police Injury Crash 06/08/2023 10:21:00 PM Maryland (State) SANDY SPRING RD County BIRMINGHAM DR NULL N/A NULL ANGLE MEETS LEFT TURN RAINING WET DARK LIGHTS ON NO CONTROLS NONE DETECTED NULL 30800813-340C-4A1A-BEB8-A86BBB4B650B No SUSPECTED MINOR INJURY RAIN, SNOW, WET NOT DISTRACTED MD C6B6C0F9-4513-4881-B04A-A796EA0E835A DESTROYED TWELVE OCLOCK TWELVE OCLOCK PASSENGER CAR MOVING CONSTANT SPEED West West 45 No No 2016 

**Classify incidents by the severity level ('Low', 'Medium', 'High') based on the speed limit.**

In [63]:
Select "Speed Limit",
    CASE
        WHEN "Speed Limit"<30 THEN 'Low'
        WHEN "Speed Limit" BETWEEN 30 and 50 THEN 'Medium'
        ELSE 'High'
    End as Severity_Level
From Crash_Reporting
LIMIT 20;


(20 row(s) affected)

Total execution time: 00:00:00.004

Speed Limit severity_level 25 Low 40 Medium 35 Medium 40 Medium 45 Medium 30 Medium 35 Medium 50 Medium 40 Medium 50 Medium 35 Medium 30 Medium 35 Medium 15 Low 35 Medium 35 Medium 15 Low 40 Medium 40 Medium 5 Low

**Calculate the average and maximum speed limit by hour of the day.**

In [29]:
SELECT EXTRACT(HOUR FROM TO_TIMESTAMP("Crash Date/Time",'MM-DD-YYYY HH24:MI:SS')) AS Hourofday,
    AVG("Speed Limit") as Average_SpeedLimit,
    MAX("Speed Limit") as Max_SpeedLimit
FROM Crash_Reporting
GROUP BY Hourofday
ORDER by Hourofday;

(12 row(s) affected)

Total execution time: 00:00:01.043

hourofday average_speedlimit max_speedlimit 1 31.5173590875031221 65 2 31.6261629772216875 65 3 32.1199864038069341 75 4 32.4483875320469572 70 5 33.4185530130142478 70 6 33.6311961371554306 65 7 33.3405762590784617 70 8 33.1997023809523810 70 9 32.8962239063954261 70 10 32.3006767910201387 65 11 31.5103547784756408 65 12 31.4346128822381262 65

**Calculate Vehicle Make and the model with there Frequency ?**

In [12]:
SELECT "Vehicle Make", "Vehicle Model", COUNT(*) AS Frequency
FROM Crash_Reporting
GROUP BY "Vehicle Make", "Vehicle Model"
ORDER BY Frequency DESC
Limit 15;


(15 row(s) affected)

Total execution time: 00:00:00.225

Vehicle Make Vehicle Model frequency TOYOTA CAMRY 5123 HONDA CIVIC 4979 HONDA ACCORD 4895 TOYOTA COROLLA 4656 UNKNOWN UNKNOWN 3646 TOYT 4S 2466 HONDA CRV 2379 NISSAN ALTIMA 2175 TOYOTA RAV4 2109 FORD TK 1912 HOND 4S 1608 FORD EXPLORER 1427 DODGE CHARGER 1419 TOYOTA PRIUS 1314 TOYOTA HIGHLANDER 1260

## **Insights** 

**Data Span and Volume:**

**Dataset contains 171,420 entries spanning January 1, 2015, to December 20, 2023.**

**Peak recorded crashes were in 2016, whereas 2020 noted significantly fewer incidents.**

**Municipalities Analysis:**

**15,867 null entries in municipality records.**

**Brookeville municipality had the lowest crash incidence from 2015 to 2023.**

**Collision Types and Crime Categories:**

**Rear-end collisions accounted for 55,557 crashes, the most frequent type.**

**'Property Damage Crash' dominated at 108,983 instances, contrasting with 423 'Fatal Crash' records.**

**Driver Distraction and Weather Impact:**

**Driver distraction noted in 106,317 records, ranging from external factors to cellphone use.**

**Weather-wise, rainy conditions contributed to 20,478 incidents, while clear weather was dominant at 116,365 crashes. Foggy conditions recorded 668 crashes.**

**Vehicle Models and Month-wise Analysis:**

**Specific vehicles like Toyota Camry (5,123) and Toyota Corolla (4,656) were highly involved in crashes.**

**October experienced the highest crash frequency (16,093), followed closely by December (14,521).**

**Incomplete Records and Lighting Conditions:**

**1,482 null entries in records, potentially indicating data gaps.**

**706 crashes were unassigned to specific counties.**

**Majority of crashes occurred in daylight (115,827) compared to low-light conditions (39,287).**